lens/s3repo/store.go

```diff
@@ -0,0 +1,116 @@
+package s3repo
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/lib/blockstore"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+)
+
+var log = logging.Logger("sql")
+
+type S3Blockstore struct {
+	prefix string
+	client http.Client
+}
+
+func NewBlockStore(connstr string) (blockstore.Blockstore, error) {
+
+	sbs := &S3Blockstore{
+		prefix: connstr,
+		client: http.Client{},
+	}
+
+	// we do not currently use the Identity codec, but just in case...
+	return blockstore.WrapIDStore(sbs), nil
+}
+
+func (sbs *S3Blockstore) Has(c cid.Cid) (has bool, err error) {
+	resp, err := sbs.client.Head(sbs.prefix + c.String() + "/data.raw")
+	if err != nil {
+		return false, err
+	}
+	return resp.StatusCode == 200, nil
+}
+
+func (sbs *S3Blockstore) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
+	return nil, fmt.Errorf("Not implemented")
+}
+
+func (sbs *S3Blockstore) Get(c cid.Cid) (blocks.Block, error) {
+	resp, err := sbs.client.Get(sbs.prefix + c.String() + "/data.raw")
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	buf, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return blocks.NewBlockWithCid(buf, c)
+}
+
+func (sbs *S3Blockstore) GetSize(c cid.Cid) (size int, err error) {
+	resp, err := sbs.client.Head(sbs.prefix + c.String() + "/data.raw")
+	if err != nil {
+		return -1, err
+	}
+	if resp.StatusCode == 200 {
+		return int(resp.ContentLength), nil
+	}
+	return -1, fmt.Errorf("does not exist")
+}
+
+func (sbs *S3Blockstore) getMasterTsKey(ctx context.Context, lookback int) (*types.TipSetKey, error) {
+	resp, err := sbs.client.Get(sbs.prefix + "/head")
```

Comment:
This didn't work when I tried fetching the url manually

Reply:
@iand note that the s3 bucket is frozen at 230700
more this weekend